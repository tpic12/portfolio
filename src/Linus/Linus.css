.links a {
  text-decoration: none;
  color: rgb(62, 72, 117);
}

.links a:hover {
  font-weight: 700;
  color: rgb(189, 89, 52);
}

.linus-section {
  width: 100vw;
  min-height: 100vh;
  background-color: white;
  color: rgb(62, 72, 117);
  display: flex;
  flex-direction: column;
  /* margin-top: 100px; */
  justify-content: flex-start;
  align-items: center;
  padding: 0 15px;
}

.linus-section h2 {
  padding-top: 10px;
  padding-bottom: 5px;
  width: 75px;
  text-align: center;
  font-family: 'Josefin Slab', serif;
  border-top: 1px solid rgb(62, 72, 117);
  border-bottom: 1px solid rgb(62, 72, 117);
  border-radius: 5px;
}

.links p {
  padding: 0;
}

.linus-circle {
  width: 100vw;
  height: 375px;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  border: 7px solid #ededed;
  border-radius: 200px;
  margin: 0 10px;
  background-color: rgb(62, 72, 117);
  color: white;
  text-align: center;
  padding: 10px 35px;
}

.linus-circle button {
  background-color: transparent;
  border: none;
  cursor: pointer;
}

.linus-circle button:hover {
  box-shadow: 0px 0px 15px #ededed;
  border-radius: 10px;
}

.linus-circle img {
  width: 100px;
  background-color: white;
  border-radius: 50%;
  animation: pulse 1s linear;
  animation-iteration-count: infinite;
}

.tech-logos {
  width: 50%;
  margin-top: 10px;
  display: flex;
  justify-content: space-evenly;
}

@media screen and (min-width: 500px) {
  .linus-circle {
    font-size: 20px;
  }

  .linus-section h2 {
    font-size: 30px;
    border-top: 2px solid rgb(62, 72, 117);
    border-bottom: 2px solid rgb(62, 72, 117);
  }
}

@media screen and (min-width: 720px) {
  .linus-circle {
    max-width: 900px;
    font-size: 24px;
  }
}