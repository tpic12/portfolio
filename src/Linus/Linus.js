import React, { Component } from "react";
import Slides from "../Slides/slides";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import Logo from "../Images/linus-logo.png";
import Linus0 from "../Images/linus0.png";
import Linus1 from "../Images/linus1.png";
import Linus2 from "../Images/linus2.png";
import Linus3 from "../Images/linus3.png";
import "./Linus.css";

class Linus extends Component {
  state = {
    slideshow: false,
    imageArr: [Linus0, Linus1, Linus2, Linus3],
  };

  handlesSlideshow = () => {
    const { slideshow } = this.state;
    this.setState({ slideshow: !this.state.slideshow });
    document.documentElement.style.overflow = !slideshow ? "hidden" : "scroll";
    document.body.scroll = !slideshow;
  };
  render() {
    const { slideshow } = this.state;
    return (
      <section id="linus-section" className="linus-section snap">
        {slideshow && (
          <Slides
            slideshow={this.handlesSlideshow}
            image={this.state.imageArr}
          />
        )}
        <h2>Linus</h2>
        <div className="links">
          <p>
            <a
              href="https://linus-ten.now.sh/"
              target="_blank"
              rel="noopener noreferrer"
            >
              Live Link
            </a>{" "}
            |{" "}
            <a
              href="https://github.com/tpic12/Linus-Client"
              target="_blank"
              rel="noopener noreferrer"
            >
              GitHub
            </a>
          </p>
        </div>
        <div className="linus-circle">
          <div className="logo-arrows">
            <a href="#qtrail-section">
              <FontAwesomeIcon icon="chevron-left" size="lg" />
            </a>
            <button onClick={this.handlesSlideshow}>
              {!slideshow && <img src={Logo} alt="linus-logo" />}
            </button>
            <a href="#spanimals-section">
              <FontAwesomeIcon icon="chevron-right" size="lg" />
            </a>
          </div>
          <p>
            Linus is a personal bartender for your home. With Linus you can add
            your favorite and current spirits in the liquor cabinet to your
            digital cabinet, and Linus will show you what cocktails you can make
            with your spirits and how! You will also be able to search for
            recipes by spirit type and make a shortlist of your favorite ones to
            try later.
          </p>
        </div>
        <div className="tech-logos">
          <FontAwesomeIcon icon={["fab", "react"]} size="2x" />
          <FontAwesomeIcon icon={["fab", "css3"]} size="2x" />
          <FontAwesomeIcon icon={["fab", "node-js"]} size="2x" />
          <FontAwesomeIcon icon={["fas", "database"]} size="2x" />
        </div>
        <div className="extra-tech">
          <ul>
            <li>Express</li>
            <li>Mocha</li>
            <li>Chai</li>
            <li>Jest</li>
          </ul>
        </div>
      </section>
    );
  }
}

export default Linus;
