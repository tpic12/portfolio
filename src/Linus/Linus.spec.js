import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Linus from "./Linus";

describe("<Landing />", () => {
  it("renders by default", () => {
    const wrapper = shallow(<Linus />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
