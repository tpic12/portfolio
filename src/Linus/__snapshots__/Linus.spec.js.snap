// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> renders by default 1`] = `
<section
  className="linus-section snap"
  id="linus-section"
>
  <h2>
    Linus
  </h2>
  <div
    className="links"
  >
    <p>
      <a
        href="https://linus-ten.now.sh/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Live Link
      </a>
       
      |
       
      <a
        href="https://github.com/tpic12/Linus-Client"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub
      </a>
    </p>
  </div>
  <div
    className="linus-circle"
  >
    <div
      className="logo-arrows"
    >
      <a
        href="#qtrail-section"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="chevron-left"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
      <button
        onClick={[Function]}
      >
        <img
          alt="linus-logo"
          src="linus-logo.png"
        />
      </button>
      <a
        href="#spanimals-section"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="chevron-right"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size="lg"
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
    </div>
    <p>
      Linus is a personal bartender for your home. With Linus you can add your favorite and current spirits in the liquor cabinet to your digital cabinet, and Linus will show you what cocktails you can make with your spirits and how! You will also be able to search for recipes by spirit type and make a shortlist of your favorite ones to try later.
    </p>
  </div>
  <div
    className="tech-logos"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fab",
          "react",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fab",
          "css3",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fab",
          "node-js",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Array [
          "fas",
          "database",
        ]
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="2x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="extra-tech"
  >
    <ul>
      <li>
        Express
      </li>
      <li>
        Mocha
      </li>
      <li>
        Chai
      </li>
      <li>
        Jest
      </li>
    </ul>
  </div>
</section>
`;
